### PR TITLE
558 filter events by location

### LINKED ISSUE #558: [feature request] filter events by location
**Is your feature request related to a problem? Please describe.**
Ability to filter events by location

**Describe the solution you'd like**
Build on top of the tag filtering system #557 , the possibility to filter by location.

**Build suggestion**
Use the `location.name` field as an other filter on the current event array.

### PR DESCRIPTION
**Pull request by BenGamma**
### created
* LocationSearch component to event list page to allow filtering by location (still need some UI modifications, will be done in 0.6 release)
### updated
* add method `onClear` as props to LocationSearch component to listen to onclear event on field.

close #558

### CODE REVIEW

File: src/stores/Events/events.store.tsx [hunk 1]
```diff
@@ -38,10 +49,31 @@ export class EventStore extends ModuleStore {
...
+    this.selectedLocation = {
+      name: '',
+      country: '',
+      countryCode: '',
+      administrative: '',
+      latlng: {
+        lat: 0,
+        lng: 0,
+      },
+      postcode: '',
```
Comment (chrismclarke) [inline src/stores/Events/events.store.tsx]: Do we need postcode? I would suggest as not all countries use postal codes might be easier to leave out and depend on name/latlng for geolocation

File: src/components/LocationSearch/LocationSearch.tsx [hunk 2]
```diff
@@ -83,6 +85,10 @@ export class LocationSearch extends React.Component<IProps, IState> {
     this.props.onChange(_resultToLocation(selected))
   }
 
+  onClearInput() {
```
Comment (chrismclarke) [inline src/components/LocationSearch/LocationSearch.tsx]: This feels redundant, we can just let `places.on('clear')` call the props onclear method directly

File: src/stores/Events/events.store.tsx [hunk 3]
```diff
@@ -29,7 +32,15 @@ export class EventStore extends ModuleStore {
   }
 
   @computed get filteredEvents() {
-    return this.filteredCollectionByTags(this.upcomingEvents, this.selectedTags)
+    return this.selectedLocation.value !== ''
+      ? this.filteredCollectionByTags(
```
Comment (chrismclarke) [inline src/stores/Events/events.store.tsx]: This doesn't read very naturally. `filteredCollection` implies the list has already been filtered, so instead this should be `filterCollection` to indicate it is an action that will happen (past tense noun vs verb). 

Also nesting one filter within another is very confusing. Move the inside filter out, give it a variable name and use in the second filter operation.

File: src/stores/Events/events.store.tsx (hunk 1 above)
Comment (BenGamma) [reply src/stores/Events/events.store.tsx]: Would make sense in this case. But I know in the user profil page we will display the complete adress, which may contain a postcode

File: src/stores/Events/events.store.tsx (hunk 3 above)
Comment (BenGamma) [reply src/stores/Events/events.store.tsx]: I put the first filter in a const, remove the ternaries and used and good old if / else. Way more readable I agree.

File: src/stores/Events/events.store.tsx (hunk 1 above)
Comment (chrismclarke) [reply src/stores/Events/events.store.tsx]: ok, more `postcode ? ... : ... ` statements it is then.

File: src/stores/Events/events.store.tsx (hunk 3 above)
Comment (chrismclarke) [reply src/stores/Events/events.store.tsx]: yeah looks a lot better, nice

File: src/components/LocationSearch/LocationSearch.tsx (hunk 2 above)
Comment (chrismclarke) [reply src/components/LocationSearch/LocationSearch.tsx]: can see this is now gone (+1)

**Pull request merged**